modules/uhttp/http.go

```diff
@@ -111,30 +117,39 @@ func newModule(
 		log.Error("Error loading http module configuration", zap.Error(err))
 	}
 	module := &Module{
-		Host:     host,
-		handlers: addHealth(getHandlers(host)),
-		mcb:      defaultInboundMiddlewareChainBuilder(log, host.AuthClient(), newStatsClient(host.Metrics())),
-		config:   cfg,
-		log:      log,
+		Host:       host,
+		router:     router,
+		handlers:   addHealth(getHandlers(host)),
+		authClient: host.AuthClient(),
+		stats:      newStatsClient(host.Metrics()),
+		config:     cfg,
+		log:        log,
 	}
-	module.mcb = module.mcb.AddMiddleware(moduleOptions.inboundMiddleware...)
 	return module, nil
 }
 
 // Start begins serving requests over HTTP
 func (m *Module) Start() error {
-	mux := http.NewServeMux()
-	// Do something unrelated to annotations
-	router := NewRouter(m.Host)
-
-	mux.Handle("/", router)
+	serveMux := http.NewServeMux()
+	serveMux.Handle("/", m.router)
 
 	for _, h := range m.handlers {
-		router.Handle(h.Path, m.mcb.Build(h.Handler))
+		// TODO: find better way of chaining the middleware
+		handle :=
```

Comment:
We can create all these handles in module constructor, and remove almost all fields from the `Module` struct.